src/common.h

```diff
@@ -13,6 +13,12 @@
 
 #include "config.h"
 
+// GAP uses various POSIX functions, such as siglongjmp
+// This is required when compiling with -std=c++11 on Cygwin
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 1
```

Comment:
It seems we require newer POSIX: snprintf for example:
```suggestion
#define _POSIX_C_SOURCE 200112L
```
But even then it breaks on macOS, because now `strlcat` and `strlcpy` are hidden, yet `configure` reported their presence *sigh*. We can of course work around that, too (for starters, I now regret that I used these BSD extension functions under their BSD names - it's OK to use them, but I just should have called them `gap_strlcpy` and `gap_strlcat` and never even bothered to check for a system implementation).

But that "easy" change is spiraling quite a bit out of control now...

Reply:
Oh and I also worry that we might be breaking yet moreL by #defining this in common.h., it'll affected every GAP kernel extension, and then those might be broken because features they rely on are "lost".

So perhaps we should just ask all GAP kernel extensions to not use the "strict" C99 / C++11 / etc. modes? Ah well